packages/taiko-client/driver/chain_syncer/blob/syncer.go

```diff
@@ -403,6 +403,165 @@ func (s *Syncer) insertNewHead(
 	return payload, nil
 }
 
+func (s *Syncer) MoveTheHead(
+	ctx context.Context,
+	txList []*types.Transaction,
+) error {
+
+	lastInsertedBlockHeader, err := s.rpc.L1.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get latest block ID from L1: %w", err)
+	}
+
+	// Ignore those already inserted blocks.
+	// if s.lastInsertedBlockID != nil && event.BlockId.Cmp(s.lastInsertedBlockID) <= 0 {
+	if s.lastInsertedBlockID != nil && lastInsertedBlockHeader.Number.Cmp(s.lastInsertedBlockID) <= 0 {
+		return nil
+	}
+
+	// Fetch the L2 parent block, if the node is just finished a P2P sync, we simply use the tracker's
+	// last synced verified block as the parent, otherwise, we fetch the parent block from L2 EE.
+	var (
+		parent *types.Header
+	)
+	if s.progressTracker.Triggered() {
+		parent, err = s.rpc.L2.HeaderByHash(ctx, s.progressTracker.LastSyncedBlockHash())
+	} else {
+		// taking last block from L2 instead of parent based on L1 block id
+		// parent, err = s.rpc.L2ParentByBlockID(ctx, event.BlockId)
+		parent, err = s.rpc.L2.HeaderByNumber(ctx, nil)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to fetch L2 parent block: %w", err)
+	}
+
+	log.Debug(
+		"Parent block",
+		"height", parent.Number,
+		"hash", parent.Hash(),
+		"beaconSyncTriggered", s.progressTracker.Triggered(),
+	)
+
+	// try to insert a new head block to L2 EE.
+	err = s.insertNewHeadUsingDecodedTxList(
+		ctx,
+		parent,
+		s.state.GetHeadBlockID(),
+		txList,
+		&rawdb.L1Origin{
+			BlockID:       lastInsertedBlockHeader.Number, // event.BlockId
+			L2BlockHash:   common.Hash{},                  // Will be set by taiko-geth.
+			L1BlockHeight: lastInsertedBlockHeader.Number, // new(big.Int).SetUint64(event.Raw.BlockNumber),
+			L1BlockHash:   lastInsertedBlockHeader.Hash(), // event.Raw.BlockHash,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to insert new head to L2 execution engine: %w", err)
+	}
+
+	metrics.DriverL1CurrentHeightGauge.Set(float64(lastInsertedBlockHeader.Number.Uint64())) //(float64(event.Raw.BlockNumber))
+	s.lastInsertedBlockID = lastInsertedBlockHeader.Number                                   //event.BlockId
+
+	if s.progressTracker.Triggered() {
+		s.progressTracker.ClearMeta()
+	}
+
+	return nil
+}
+
+// insertNewHead tries to insert a new head block to the L2 execution engine's local
+// block chain through Engine APIs.
+func (s *Syncer) insertNewHeadUsingDecodedTxList(
+	ctx context.Context,
+	parent *types.Header,
+	headBlockID *big.Int,
+	txList []*types.Transaction,
+	l1Origin *rawdb.L1Origin,
+) error {
+	log.Debug(
+		"Try to insert a new L2 head block",
+		"parentNumber", parent.Number,
+		"parentHash", parent.Hash(),
+		"headBlockID", headBlockID,
+		"l1Origin", l1Origin,
+	)
+
+	// Get L2 baseFee
+	baseFeeInfo, err := s.rpc.TaikoL2.GetBasefee(
+		&bind.CallOpts{BlockNumber: parent.Number, Context: ctx},
+		l1Origin.BlockID.Uint64(), // event.Meta.L1Height,
```

Comment:
using block id instead of L1 height

Reply:
i would just get l1 height from l1 execution client instead.